### ISSUE TITLE
TingoDB support

### ISSUE THREAD

**Issue by Phoscur** [kcbanner/connect-mongo#179]
There is an outdated module already supporting this, it's derived from here.
[https://github.com/FrankHassanabad/connect-tingo/issues/3]

Although connect-mongo and connect-tingo are not very different, I can't make out the reason why connect-mongo fails silently with a TingoDB instance [session is not created and requests are not delivered]).

Would it be possible to add TingoDB support to connect-mongo? It would be nice to have one less db engine switch!

**Comment by jdesboeufs** [comment]
You are free to submit a PR for that :)
I've no time to investigate the problem for now, but `connect-mongo` has diverged since many months.
Can you try to pass a TingoDB `Db` instance to `connect-mongo` `db` option?

**Comment by jdesboeufs** [comment]
The problem may have been fixed in upcoming version (1.0.0).

**Comment by muka** [comment]
The issue is at the check for `options.db.listCollections`  that tingodb does not have. It has a `_path` property instead (https://github.com/kcbanner/connect-mongo/blob/master/src/connect-mongo.js#L88)

**Comment by kgividen** [comment]
FYI This worked for me with tingodb so it would be nice if this PR was accepted.

**Comment by jdesboeufs** [comment]
TingoDB is not officially supported by connect-mongo.
If someone want to fix that, I'll be happy to accept a PR.

**Comment by CodeLenny** [comment]
@muka @jdesboeufs I'm getting a new error in 1.3.x (seen in both 1.3.0 and 1.3.1) even with the workaround.  1.2.1 was working well with the workaround in place.

```
TypeError: Cannot read property 'upserted' of undefined
  at <PROJECT>/node_modules/connect-mongo/src/index.js:249:43
  at tryCatcher (<PROJECT>/node_modules/bluebird/js/release/util.js:16:23)
  at Promise._settlePromiseFromHandler (<PROJECT>/node_modules/bluebird/js/release/promise.js:504:31)
  at Promise._settlePromise (<PROJECT>/node_modules/bluebird/js/release/promise.js:561:18)
  at Promise._settlePromise0 (<PROJECT>/node_modules/bluebird/js/release/promise.js:606:10)
  at Promise._settlePromises (<PROJECT>/node_modules/bluebird/js/release/promise.js:685:18)
  at Promise._fulfill (<PROJECT>/node_modules/bluebird/js/release/promise.js:630:18)
  at <PROJECT>/node_modules/bluebird/js/release/nodeback.js:42:21
  at <PROJECT>/node_modules/tingodb/lib/wqueue.js:37:7
  at <PROJECT>/node_modules/tingodb/lib/tcoll.js:752:6
  at _fnApply (<PROJECT>/node_modules/safe/lib/safe.js:172:22)
  at _catcher (<PROJECT>/node_modules/safe/lib/safe.js:349:11)
  at <PROJECT>/node_modules/safe/lib/safe.js:462:4
  at _wrapArgsSure (<PROJECT>/node_modules/safe/lib/safe.js:218:14)
  at <PROJECT>/node_modules/safe/lib/safe.js:332:13
  at task (<PROJECT>/node_modules/safe/lib/safe.js:642:8)
  at <PROJECT>/node_modules/safe/lib/safe.js:656:8
  at <PROJECT>/node_modules/safe/lib/safe.js:342:13
  at <PROJECT>/node_modules/tingodb/lib/wqueue.js:37:7
  at <PROJECT>/node_modules/tingodb/lib/tcoll.js:601:4
  at <PROJECT>/node_modules/safe/lib/safe.js:342:13
  at <PROJECT>/node_modules/tingodb/lib/tcoll.js:589:6
  at _fnApply (<PROJECT>/node_modules/safe/lib/safe.js:174:22)
  at _catcher (<PROJECT>/node_modules/safe/lib/safe.js:349:11)
  at <PROJECT>/node_modules/safe/lib/safe.js:462:4
  at _wrapArgsSure (<PROJECT>/node_modules/safe/lib/safe.js:218:14)
  at FSReqWrap.wrapper [as oncomplete] (fs.js:667:5)
```

**Comment by jdesboeufs** [comment]
@CodeLenny It will be fixed in `1.3.2`. I'm publishing it right now!

**Comment by CodeLenny** [comment]
@jdesboeufs Great, thanks so much!
